### PR TITLE
Add Group Component

### PR DESCRIPTION
**Pull request by darenschaad**
Groups component with mock data is being displayed in the admin tab.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -31,6 +31,18 @@ dev_build:
...
+	echo $@
+	docker-compose -f docker-compose.dev.yml up
+
+dev_down:
+	echo $@
+	docker-compose -f docker-compose.dev.yml down
+
+dev_build:
+	echo @
+	docker-compose -f docker-compose.dev.yml build
```
Comment (maxkwallace) [inline Makefile]: These changes should possibly be in a separate PR (if they're not already) since they're not related to the Groups component.

cc @wittejm

File: src/frontend/src/redux/groups/actions.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
+import { Dispatch } from 'redux';
+import { LOAD_GROUPS, Group } from './types';
+
+export const loadGroups = () => (dispatch: Dispatch) => {
+  return new Promise(resolve => {
```
Comment (maxkwallace) [inline src/frontend/src/redux/groups/actions.ts]: Not that it really matters since this code will be replaced with the actual request, but since I thought of it lol the Promise should probably be resolved.

File: src/frontend/src/components/GroupList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import { connect } from 'react-redux';
+import { AppState } from '../../redux/store';
+import { loadGroups } from '../../redux/groups/actions';
+import { Group as GroupTypes, GroupState } from '../../redux/groups/types';
+import Group from '../Group';
+
+interface Props {
+    groups: GroupState;
+    loadGroups: Function;
+}
```
Comment (maxkwallace) [inline src/frontend/src/components/GroupList/index.tsx]: The indentation is off in this file. Should be 2 spaces, same as the others.

File: Makefile (hunk 1 above)
Comment (wittejm) [reply Makefile]: That's a git behavior I don't really get, because the changes from mine and Matt's commits were already merged into Master. @darenschaad, maybe you could remove those commits from your branch using `git rebase -i <base-branch>` or some other git-fu?

Comment (ForrestLonganecker) [reply Makefile]: I have had this issue in the past and it had to do with some things being merged into my local master branch that were then for some reason seen as additions to the remote master when I tried to merge the branch back into the remote eventually.

If you are still having issues with this tomorrow I'll be at the meetup and can help puzzle it out!

Comment (NickSchimek) [reply Makefile]: This can occur when you merge in changes from another branch. This should be able to be fixed with a rebase. I’d be happy to walk anyone through the process. There are also a couple videos I posted in the #random slack channel that shows how to rebase.

Comment (NickSchimek) [reply Makefile]: To avoid this in the future I’d advise to rebase on top of master instead of pulling from master. If interested I can walk you through a workflow that I find helpful.

**Comment by ForrestLonganecker** [comment]
Hey @darenschaad how is this PR coming along? Where you able to remove the code that was coming from someone else's PR? Let me know if there is anything I can do to help!

**Comment by NickSchimek** [comment]
@darenschaad Are you still working on this PR?

**Comment by NickSchimek** [comment]
@ForrestLonganecker FYI: I got a hole of Daren through slack. He is still working on this, and should be at the next meetup to finish it up.

**Comment by ForrestLonganecker** [comment]
Thanks for the update!
